### ISSUE TITLE
Remove psalm errors

### ISSUE THREAD

**Issue by christianlupus** [nextcloud/cookbook#1706]
Currently, there are quite done psalm issues detected and hidden by the baseline. These issues should be fixed by using correct annotations. 

It might be questionable is the baseline should be updated upon pr merging automatically.

**Comment by christianlupus** [comment]
See also #990

**Comment by provokateurin** [comment]
If you don't mind I can also fix the rest which is only stubs AFAIK.

**Comment by christianlupus** [comment]
I think 2 classes (`oc_image` and `OC\Hooks\Emitter`) cause trouble. So, you suggest to copy the required class methods into files in the cookbook source code without actual content and located such that these are not interfering with the DI algorithm?

**Comment by provokateurin** [comment]
Yeah basically that. I've done it for a few apps now, so it's relatively easy for me. Besides those classes, often doctrine and symfony classes are needed as well because they are used somewhere.
